### PR TITLE
Add example extractors

### PR DESCRIPTION
**Pull request by yenicelik**
### Implemented and tested these extractors: 

- Invoice Extractor (using Donut)
- Language Extractor (provided full-text, chunks the text and predicts the language for each chunk)
- Identity Extractor (hashes the input-bytes into a vector using sha256, useful for quick duplicate-checks)

### How to reproduce:
(1) Package the extractor into a docker image `cargo run extractor package --dev -v --config-path extractors/simple_invoice_parser.yaml`
(2) Run the extractor, mounting any files required 
```
docker run \
--mount type=bind,source="$(pwd)/data",target=/indexify/data \ 
-it \
--rm \
-e RUST_LOG=debug \ 
yenicelik/simple-invoice-parser extractor extract \ 
--file data/...
```
`cargo run` can be replaced by `./target/debug/indexify` or `indexify` if the binaries were added to the `PATH`.

### For maintainers & contributors:
These steps are usually not needed when working only extractor! Currently, the extractor-base image is pulled from DockerHub, due to how BuildKit works (it does not use a local registry). Unfortunately, it seems tedious to resolve this (see https://github.com/moby/buildkit/issues/2343). If you need to modify the rust code to run the extractors, please run 
- create an account for dockerhub
- modify the Makefile credentials at the top of the Makefile
- run `make build-base-extractor-push`, this will build the image, and push it to dockerhub
- modify `dockerfiles/Dockerfiles.extractor` to use your own `FROM ...`, (i.e. `FROM yenicelik/indexify-extractor-base` instead `FROM diptanu/indexify-extractor-base`).
- Then you can go to the "How to reproduce" section above

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu
+FROM ubuntu:22.04
```
Comment (diptanu) [inline .devcontainer/Dockerfile:1]: Can we move this dev container change into a separate PR?

File: docs/docs/develop.md [hunk 2]
```diff
@@ -97,3 +101,6 @@ Visual Studio Code Devcontainers have been setup as well. Opening the codebase i
 ### Docker-compose
 
 If you're within the dev container, you can call the docker-compose-v1 from within /usr/bin/
+
+If docker produces a EONET error, please try to build your devcontainer prior to launching it in vscode:
```
Comment (diptanu) [inline docs/docs/develop.md:105]: Move this into the dev container pr please.

File: extractors/identity_hash_embedding.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    def __init__(self):
+        print("Hello")
+        super(IdentityHashEmbedding, self).__init__(max_context_length=128)
+        self._model = hashlib.sha256()
+
+    def extract_embeddings(self, texts: List[str]) -> List[List[float]]:
+        return [self._embed(text) for text in texts]
+
+    def extract_query_embeddings(self, query: str) -> List[float]:
```
Comment (diptanu) [inline extractors/identity_hash_embedding.py:33]: No longer needed.

File: extractors/identity_hash_embedding.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+name: yenicelik/identity-hash-extractor
```
Comment (diptanu) [inline extractors/identity_hash_embedding.yaml:1]: Can we move the extractor into a separate PR ?

File: extractors/language_extractor.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            data = {"language": language.name, "score": str(confidence)}
+            out.append(
+                [Content.from_text(
+                    text=content_texts[i],
+                    feature=Feature.metadata(value=data, name="language"),
+                )]
+            )
+        return out
+
+    def extract_query_embeddings(self, query: str) -> List[float]:
```
Comment (diptanu) [inline extractors/language_extractor.py:45]: No longer needed

File: setup.py [hunk 6]
```diff
@@ -50,7 +50,7 @@
         "pytest-asyncio",
         "parameterized",
     ],
-    "test": ["pypdf", "span_marker", "parameterized"],
+    "test": ["pypdf", "span_marker", "lingua-language-detector", "parameterized"],
```
Comment (diptanu) [inline setup.py:53]: Do we need this anymore? If extractors are moving to their own folder, developers can put dependencies into a requirements.txt no? We can remove all these test dependencies out of here.

File: src/extractors.rs [hunk 7]
```diff
@@ -208,7 +208,8 @@ pub fn run_extractor(extractor_path: Option<String>, text: Option<String>, file_
         }
         (None, Some(file_path)) => {
             let data = std::fs::read(file_path)?;
-            let content = PyContent::from_bytes(data, internal_api::ContentType::Text).try_into()?;
+            let content =
```
Comment (diptanu) [inline src/extractors.rs:211]: Unrelated change can we remove this from the commit ?

**Review by diptanu — CHANGES_REQUESTED** [review]
Looks great overall! Thanks David <3 
The main comment is to break down the PR into many commits, and move some code around.

File: src/extractors.rs (hunk 7 above)
Comment (yenicelik) [reply src/extractors.rs]: reset it to the state of main branch now

File: setup.py (hunk 6 above)
Comment (yenicelik) [reply setup.py:53]: Hmm this was needed for the cargo tests, but redundant now, I will remove this

**Comment by yenicelik** [comment]
Please make sure to merge https://github.com/diptanu/indexify/pull/146 before this

**Comment by yenicelik** [comment]
Will move the extractors here https://github.com/tensorlakeai/indexify-extractors and remove them from diptanu/indexify

**Pull request closed without merging**